### PR TITLE
Hotfix/wan tcp [4322]

### PR DESCRIPTION
**Pull request by MiguelBarro**
#4279
- Fast RTPS class TCPv4Transport modified to proper WAN operation.
- HelloWorldTCP example WAN enabled.

### CODE REVIEW

File: examples/C++/HelloWorldExampleTCP/HelloWorldPublisher.cpp [hunk 1]
```diff
@@ -59,7 +59,12 @@ bool HelloWorldPublisher::init()
     descriptor->sendBufferSize = 0;
     descriptor->receiveBufferSize = 0;
     //descriptor->set_WAN_address("127.0.0.1");
-    descriptor->add_listener_port(5100);
+	if (!wan_ip.empty())
+	{
+		descriptor->set_WAN_address(wan_ip);
+		std::cout << wan_ip << ":" << port << std::endl;
+	}
```
Comment (LuisGP) [inline examples/C++/HelloWorldExampleTCP/HelloWorldPublisher.cpp]: Please, change tabs to 4 spaces.

File: examples/C++/HelloWorldExampleTCP/HelloWorldSubscriber.cpp [hunk 2]
```diff
@@ -36,15 +36,23 @@ mp_subscriber(nullptr)
...
-    initial_peer_locator.port = 5100;
+	if (!wan_ip.empty())
+	{
+		IPLocator::setIPv4(initial_peer_locator, wan_ip);
+		std::cout << wan_ip << ":" << port << std::endl;
+	}
+	else
+	{
+		IPLocator::setIPv4(initial_peer_locator, "127.0.0.1");
+	}
```
Comment (LuisGP) [inline examples/C++/HelloWorldExampleTCP/HelloWorldSubscriber.cpp]: Please, change tabs to 4 spaces.

File: examples/C++/HelloWorldExampleTCP/HelloWorld_main.cpp [hunk 3]
```diff
@@ -34,6 +36,8 @@ int main(int argc, char** argv)
     int type = 1;
     int count = 0;
     long sleep = 100;
+	std::string wan_ip;
+	int port = 5100;
```
Comment (LuisGP) [inline examples/C++/HelloWorldExampleTCP/HelloWorld_main.cpp]: Please, change tabs to 4 spaces.

File: examples/C++/HelloWorldExampleTCP/HelloWorld_main.cpp [hunk 4]
```diff
@@ -43,24 +47,46 @@ int main(int argc, char** argv)
...
+
+				if (argc > 4)
+				{
+					wan_ip = std::string(argv[4]);
+				}
+
+				if (argc > 5)
+				{
+					port = atoi(argv[5]);
+				}
```
Comment (LuisGP) [inline examples/C++/HelloWorldExampleTCP/HelloWorld_main.cpp]: Please, change tabs to 4 spaces.

File: examples/C++/HelloWorldExampleTCP/HelloWorld_main.cpp [hunk 5]
```diff
@@ -43,24 +47,46 @@ int main(int argc, char** argv)
...
+			
+			if (argc > 2)
+			{
+				wan_ip = std::string(argv[2]);
+			}
+
+			if (argc > 3)
+			{
+				port = atoi(argv[3]);
+			}
```
Comment (LuisGP) [inline examples/C++/HelloWorldExampleTCP/HelloWorld_main.cpp]: Please, change tabs to 4 spaces.

File: examples/C++/HelloWorldExampleTCP/HelloWorld_main.cpp [hunk 6]
```diff
@@ -43,24 +47,46 @@ int main(int argc, char** argv)
...
     {
         std::cout << "publisher OR subscriber argument needed" << std::endl;
-        std::cout << "publisher has optional arguments: publisher [times] [interval] " << std::endl;
+        std::cout << "publisher has optional arguments: publisher [times] [interval] [wan_ip] [port] " << std::endl;
         std::cout << "\ttimes: Number of messages to send (default: unlimited = 0). " << std::endl;
         std::cout << "\t\tIf times is set greater than 0, no messages will be sent until a subscriber matches. " << std::endl;
-        std::cout << "\tinterval: Milliseconds between messages (default: 100). " << std::endl;
+		std::cout << "\tinterval: Milliseconds between messages (default: 100). " << std::endl;
+		std::cout << "\twap_ip: TODO " << std::endl;
+		std::cout << "\tport: TODO " << std::endl;
```
Comment (LuisGP) [inline examples/C++/HelloWorldExampleTCP/HelloWorld_main.cpp]: Please, change tabs to 4 spaces, and add the description of the fields. Remember in this case, the publisher will act as TCP server, so the IP is used iss WAN address and the subscriber will act as TCP client, so the IP must be the remote public address.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1075/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1045/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1059/)

File: examples/C++/HelloWorldExampleTCP/HelloWorld_main.cpp (hunk 3 above)
Comment (JuanCarlos-eProsima) [reply examples/C++/HelloWorldExampleTCP/HelloWorld_main.cpp]: Fixed

File: examples/C++/HelloWorldExampleTCP/HelloWorld_main.cpp (hunk 4 above)
Comment (JuanCarlos-eProsima) [reply examples/C++/HelloWorldExampleTCP/HelloWorld_main.cpp]: Fixed

File: examples/C++/HelloWorldExampleTCP/HelloWorld_main.cpp (hunk 5 above)
Comment (JuanCarlos-eProsima) [reply examples/C++/HelloWorldExampleTCP/HelloWorld_main.cpp]: Fixed

File: examples/C++/HelloWorldExampleTCP/HelloWorld_main.cpp (hunk 6 above)
Comment (JuanCarlos-eProsima) [reply examples/C++/HelloWorldExampleTCP/HelloWorld_main.cpp]: Updated.

File: examples/C++/HelloWorldExampleTCP/HelloWorldPublisher.cpp (hunk 1 above)
Comment (JuanCarlos-eProsima) [reply examples/C++/HelloWorldExampleTCP/HelloWorldPublisher.cpp]: Fixed

File: examples/C++/HelloWorldExampleTCP/HelloWorldSubscriber.cpp (hunk 2 above)
Comment (JuanCarlos-eProsima) [reply examples/C++/HelloWorldExampleTCP/HelloWorldSubscriber.cpp]: Fixed

**Pull request closed without merging**